src/poetry/publishing/uploader.py

```diff
@@ -275,7 +275,9 @@ def _upload_file(
                         "Redirects are not supported. "
                         "Is the URL missing a trailing slash?"
                     )
-                elif resp.status_code == 400 and "was ever registered" in resp.text:
+                elif (
```

Comment:
Seems unrelated to addressing the behavior around 3xx codes - https://github.com/python-poetry/poetry/issues/7159

Reply:
Removed this change.